packages/rum-core/src/domain/rumEventsCollection/view/trackViews.ts

```diff
@@ -120,21 +101,69 @@ export function trackViews(location: Location, lifeCycle: LifeCycle) {
     return newView(lifeCycle, location, isRecording, ViewLoadingType.ROUTE_CHANGE, currentView.url, startClocks, name)
   }
 
+  function startViewLifeCycle() {
+    lifeCycle.subscribe(LifeCycleEventType.SESSION_RENEWED, () => {
+      // do not trigger view update to avoid wrong data
+      currentView.end()
+      if (areViewsTrackedAutomatically) {
+        // Renew view on session renewal
+        currentView = trackViewChange()
+      }
+    })
```

Comment:
Has it be discussed somewhere? Since the user can't know when the session is renewed, they won't be able to create a new view in this case.

Reply:
> Has it be discussed somewhere?

No!
IMO, it seems more aligned with the manual tracking strategy to not take the responsibility to start a new view in this case. It can also be deceptive to have view without name in this case and I am not sure that we can make the assumption that we can reuse the previous view name. 
We could expose something like `DD_RUM.onSessionRestart(callback)` API to allow customer to deal with these cases.
wdyt?

Reply:
Ok it makes sense to avoid making assumption on the new view name. I would prefer to avoid adding new APIs if possible. Maybe ensuring that the session is renewed when using `DD_RUM.startView()` could be enough?